Orange/data/table.py

```diff
@@ -178,6 +192,8 @@ def Y(self):
     def Y(self, value):
         if len(value.shape) == 1:
             value = value[:, None]
+        if sp.issparse(value) and value.shape[0] == 1:
```

Comment:
Hmmm, this might be a bit more complicated. Currently, this doesn't work for tables with only one instance but multiple targets. 
```py
>>> data = Table('iris')[:1]

>>> data.X = sp.csr_matrix(data.X)
>>> multiple_targets = np.hstack((data.Y[:, None], data.Y[:, None]))
>>> data.Y = sp.csr_matrix(multiple_targets)

>>> print(data.X.shape)
(1, 4)
>>> print(multiple_targets.shape)
(1, 2)
>>> print(data.Y.shape)
(2, 1)
```
Maybe something like this will do? 
`sp.issparse(value) and len(self) != value.shape[0] and value.shape[1] == len(self)`?